async/kafka/kafka.go

```diff
@@ -72,212 +79,51 @@ func (m *message) Nack() error {
 	return nil
 }
 
-// Factory definition of a consumer factory.
-type Factory struct {
-	name    string
-	topic   string
-	group   string
-	brokers []string
-	oo      []OptionFunc
-}
-
-// New constructor.
-func New(name, topic, group string, brokers []string, oo ...OptionFunc) (*Factory, error) {
-
-	if name == "" {
-		return nil, errors.New("name is required")
-	}
-
-	if len(brokers) == 0 {
-		return nil, errors.New("provide at least one broker")
-	}
-
-	if topic == "" {
-		return nil, errors.New("topic is required")
-	}
-
-	return &Factory{name: name, topic: topic, group: group, brokers: brokers, oo: oo}, nil
-}
-
-// Create a new consumer.
-func (f *Factory) Create() (async.Consumer, error) {
+//SaramaConfig function creates a sarama config object with the default configuration
+func SaramaConfig(name string) (*sarama.Config, error) {
```

Comment:
The name of the function can explain in more detail what it does.
```suggestion
func DefaultSaramaConfig(name string) (*sarama.Config, error) {
```